lib/ansible/modules/monitoring/grafana_dashboard.py

```diff
@@ -195,6 +200,28 @@ def get_grafana_version(module, grafana_url, headers):
     return int(grafana_version)
 
 
+def grafana_folder_exists(module, grafana_url, folder_name, headers):
+    # the 'General' folder is a special case, it's ID is always '0'
+    if folder_name == 'General':
+        return True, 0
+
+    try:
+        r, info = fetch_url(module, '%s/api/folders' % grafana_url, headers=headers, method='GET')
+
+        if info['status'] != 200:
+            raise GrafanaAPIException("Unable to query Grafana API for folders (name: %s): %d" % (folder_name, info['status']))
+
+        folders = json.load(r.read())
```

Comment:
`json.loads` unless it will fail with error `'bytes' object has no attribute 'read'`

Reply:
I've tried your branch on my local grafana with the `json.loads` fix. 
If I import an existing dashboard (previously in general folder) it will be moved to the specified folder.
but the module is not idempotent anymore. If I re-run the playbook the task will return un `changed` status even if the dashboard json file has no changes.

Reply:
@seuf Can you share your "fix"? I'll have a look at the idempotent problem.

Reply:
`json.load` -> `json.loads` with a `s`